vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/FSPackageRegistry.java

```diff
@@ -138,15 +138,31 @@ public FSPackageRegistry(@NotNull File homeDir) throws IOException {
      * @throws IOException If an I/O error occurs.
      */
     public FSPackageRegistry(@NotNull File homeDir, InstallationScope scope) throws IOException {
+       this(homeDir, scope, null);
+    }
+
+    public FSPackageRegistry(@NotNull File homeDir, InstallationScope scope, @Nullable AbstractPackageRegistry.SecurityConfig securityConfig) throws IOException {
+        super(securityConfig);
         this.homeDir = homeDir;
+        log.info("Jackrabbit Filevault FS Package Registry initialized with home location {}", this.homeDir.getPath());
         this.scope = scope;
         loadPackageCache();
     }
-
     /**
      * Deafult constructor for OSGi initialization (homeDir defined via activator)
+     * @throws IOException 
      */
-    public FSPackageRegistry() {
+    @Activate
```

Comment:
since when can you use a constructor as activator?

Reply:
Since DS 1.4 (supported since AEM 6.3): https://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.component.html#service.component-constructor.injection